### PR TITLE
Update Trace Agent macrobenchmark

### PR DESCRIPTION
**Pull request by igoragoli**
### What does this PR do?

Updates the Trace Agent macrobenchmark. Changes are described in the corresponding [PR](https://github.com/DataDog/benchmarking-platform/pull/31) in the benchmarking-platform repo.

### Motivation
We are using a new set of tags in the benchmarking platform, and a new tool to execute experiments.

### Describe how to test/QA your changes
- CI Pipeline that runs one of the new macrobenchmarks: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/337473219
- Results on a testing dashboard: https://ddstaging.datadoghq.com/dashboard/vu2-f8r-kts/benchmarking-platform--trace-agent?refresh_mode=paused&tpl_var_bp_app%5B0%5D=%2A&tpl_var_bp_project%5B0%5D=%2A&tpl_var_ci_pipeline_id%5B0%5D=20592268&from_ts=1695828796235&to_ts=1695829471000&live=false
- More testing was performed in the context of the benchmarking-platform repo, please see the linked PR.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/benchmarks/macrobenchmarks.yml [hunk 1]
```diff
@@ -13,20 +13,15 @@ variables:
...
   needs: ["setup_agent_version"]
   script:
-    - export ARTIFACTS_DIR="$(pwd)/reports" && (mkdir "${ARTIFACTS_DIR}" || :)
-    - export DD_API_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.dd_api_key --with-decryption --query "Parameter.Value" --out text)
     - git clone --branch trace-agent https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab.ddbuild.io/DataDog/benchmarking-platform /platform && cd /platform
-    - ./steps/capture-hardware-software-info.sh
     - ./steps/run-benchmarks.sh
-    - "./steps/upload-results-to-s3.sh || :"
   artifacts:
     name: "reports"
```
Comment (ddyurchenko) [inline .gitlab/benchmarks/macrobenchmarks.yml:19]: Reference: https://github.com/DataDog/dd-trace-rb/blob/master/.gitlab/benchmarks.yml

For artifacts uploading, please use

```
artifacts:
    name: "artifacts"
    when: always
    paths:
      - platform/artifacts/
    expire_in: 3 months
```

File: .gitlab/benchmarks/macrobenchmarks.yml [hunk 2]
```diff
@@ -38,48 +33,56 @@ variables:
   # benchmarks get changed to run on every PR)
   allow_failure: true
 
-trace-agent-v04-4cpus-normal_load-fixed_sps-macrobenchmarks:
+trace-agent-v04-2cpus-normal_load-fixed_sps:
   extends: .trace_agent_benchmarks
   variables:
-    TRACE_AGENT_VERSION: main
     TRACE_AGENT_ENDPOINT: v04
     TRACE_AGENT_CPUS: 2
```
Comment (ddyurchenko) [inline .gitlab/benchmarks/macrobenchmarks.yml:40]: Let's discuss this in PR in benchmarking-platform repo, mb it is better to use `TRACE_AGENT_CPUS: 40-41` to simplify scripts.

**Comment by igoragoli** [comment]
## Updates
- Resolved conversations (modified job artifacts name, path and set schedule to `always`; modified `git clone` command)
- Resolved issues in the benchmarking-platform repo
 
## Testing
- Can be see in the following PR: https://github.com/DataDog/benchmarking-platform/pull/31

**Review by ddyurchenko — APPROVED** [review]
Much better!

**Pull request merged**